supplementary_style_guide/style_guidelines/general-formatting.adoc

```diff
@@ -100,5 +100,67 @@ Text for note.
 ====
 ----
 
-// [[product-names-versions-ref]]
-// == Product names and version references
+[product-names-versions-ref]]
+== Product names and version references
+
+Whenever you refer to the name of your product in full, or in its abbreviated form, or when you refer to the major and minor version of your product, avoid using hard-coded references and use attributes instead.
+Only use hard-coded version references if the version that you are referring to in a particular case never changes.
+
+=== Attribute file
+
+Define attributes for product name and product version and store them in a dedicated attributes file for each set of product documentation.
+For examples of where you can store the shared attributes file inside your documentation repository, see the link:https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example[Example modular documentation repository].
+Include the attributes file at the beginning of the `master.adoc` files of all titles in your documentation set:
+
+.master.adoc
+[source,asciidoc,options="nowrap"]
+----
+\include::<path-to-directory-with-attributes-file>/attributes.adoc[]
+----
+
+=== Minimum required attributes
+
+Define attributes for the following values in each documentation set.
+Note that the attribute names used in this section are only meant as examples.
+You can use different attribute names:
+
+The name of the product:: Use the product name attribute for all instances of the product name where possible.
+Avoid using hard-coded product names.
+For example:
++
+.Example product name attribute
+----
+:name-product: Red Hat build of Quarkus
+----
+
+The abbreviated form of the product name:: If it is necessary for your product, you can use an attribute to store a shortened version of the name of your product, for example:
++
+.Example abbreviated product name attribute
+----
+:name-product-abbreviated: RHBQ
```

Comment:
Could we use a different product for these examples? I just checked the master naming spreadsheet for this and it specifically says to not use RHBQ as a shortened form.